File: DsWorkNet/Dswork.Core/Util/EncryptUtil.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace Dswork.Core.Util
{
	/// <summary>
	/// 加密算法（MD5、Base64）
	/// </summary>
	public class EncryptUtil
	{
		/// <summary>
		/// MD5加密
		/// </summary>
		/// <param name="str">需要加密的字符串</param>
		/// <returns>32位MD5的String，失败返回null</returns>
		public static String EncryptMd5(String str)
		{
			try
			{
				byte[] result = Encoding.Default.GetBytes(str);
				MD5 md5 = new MD5CryptoServiceProvider();
				byte[] output = md5.ComputeHash(result);
				return BitConverter.ToString(output).Replace("-", "");
			}
			catch
			{
			}
			return null;
		}


		/// <summary>
		/// 将字符串转化为base64编码
		/// </summary>
		/// <param name="str">需要加密的字符串</param>
		/// <returns>base64编码的字符串，失败返回null</returns>
		public static String EncodeBase64(String str)
		{
			try
			{
				return Convert.ToBase64String(Encoding.Default.GetBytes(str));
			}
			catch
			{
				return null;
			}
		}

		/// <summary>
		/// 将 BASE64 编码的字符串 进行解码
		/// </summary>
		/// <param name="str">base64编码的字符串</param>
		/// <returns>解码后的字符串，失败返回null</returns>
		public static String DecodeBase64(String str)
		{
			try
			{
				return Encoding.Default.GetString(Convert.FromBase64String(str));
			}
			catch
			{
				return null;
			}
		}
	}
}

File: DsWorkAndroid/dswork/dswork/android/util/EncryptUtil.java
package dswork.android.util;

import java.security.MessageDigest;
import java.util.Locale;

import android.util.Base64;

/**
 * 加密算法（MD5、Base64）
 */
public class EncryptUtil
{
	/**
	 * MD5加密
	 * @param str 需要加密的String
	 * @return 32位MD5的String，失败返回null
	 */
	public static String encryptMd5(String str)
	{
		if(str != null)
		{
			StringBuilder sb = new StringBuilder();
			try
			{
				MessageDigest md = MessageDigest.getInstance("MD5");
				byte[] digest = md.digest(str.getBytes());
				String stmp = "";
				for(int n = 0; n < digest.length; n++)
				{
					stmp = (Integer.toHexString(digest[n] & 0XFF));
					sb.append((stmp.length() == 1)?"0":"").append(stmp);
				}
				return sb.toString().toUpperCase(Locale.ENGLISH);
			}
			catch(Exception e)
			{
				e.printStackTrace();
			}
			finally
			{
				sb = null;
			}
		}
		return null;
	}

	/**
	 * 将字符串转化为base64编码
	 * @param str 需要加密的String
	 * @return base64编码的String，失败返回null
	 */
	public static String encodeBase64(String str)
	{
		try
		{
			return Base64.encodeToString(str.getBytes(), Base64.CRLF);
		}
		catch(Exception e)
		{
			return null;
		}
	}

	/**
	 * 将 BASE64 编码的字符串 进行解码
	 * @param str base64编码的字符串
	 * @return 解码后的字符串，失败返回null
	 */
	public static String decodeBase64(String str)
	{
		try
		{
			if(str != null)
			{
				return new String(Base64.decode(str, Base64.CRLF));
			}
		}
		catch(Exception e)
		{
		}
		return null;
	}
}
